devito/at_controller.py

```diff
@@ -0,0 +1,344 @@
+import os
+import stat
+import math
+import shutil
+import cpuinfo
+import subprocess
+
+# global vars.
+# report folder will be created in current dir. Change if necessary
+reports_folder_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "auto-tune-reports")
```

Comment:
This should have been configurable more easily. Libraries like devito are expected to be installed to common directories to which every user does not have write access. We were using temp directories so far for all writes. Since this is the first time we are making writes to the directory that contains the devito code, I think this should be a configuration parameter outside the code.